src/core/thread/mesh_forwarder.cpp

```diff
@@ -806,12 +806,15 @@ Neighbor *MeshForwarder::UpdateNeighborOnSentFrame(Mac::TxFrame &aFrame, otError
     else if (aError == OT_ERROR_NO_ACK)
     {
         neighbor->IncrementLinkFailures();
+
+#if OPENTHREAD_FTD
         VerifyOrExit(Mle::Mle::IsActiveRouter(neighbor->GetRloc16()));
 
         if (neighbor->GetLinkFailures() >= Mle::kFailedRouterTransmissions)
         {
-            Get<Mle::MleRouter>().RemoveNeighbor(*neighbor);
+            Get<RouterTable>().RemoveNeighbor(static_cast<Router &>(*neighbor));
```

Comment:
Should we do? `Signal(OT_NEIGHBOR_TABLE_EVENT_ROUTER_REMOVED, aNeighbor);`?

---

(unrelated to this: but at some point, I'd think it'd be good to move such notification events to the router table itself).

Reply:
Note that this change simply removes the direct neighbor link (by setting the link quality to 0). Do we want to notify when the link quality transitions to 0 but the router entry has not yet been removed?

Reply:
I see. The `Neighbor` state remains as before, so it will still show up in the "neighbor table". 
Where/how would we remove this neighbor (until the router id entry is removed?)

Reply:
Router ID entry is removed or timeout occurs due to not receiving MLE Advertisements.

Reply:
Sounds good to me. 👍 I double-checked that in the case of `OPENTHREAD_CONFIG_MLE_SEND_LINK_REQUEST_ON_ADV_TIMEOUT` we don't expect for link failures to remove the neighbor and use the timeout.